lib/modules/platform/github/index.ts

```diff
@@ -1378,6 +1379,49 @@ export async function ensureIssueClosing(title: string): Promise<void> {
   }
 }
 
+async function addMilestone(
+  issueNo: number,
+  milestoneTitle: string,
+): Promise<void> {
+  logger.debug(`Adding milestone '${milestoneTitle}' to #${issueNo}`);
+  // first, find the milestone id
+  const milestones = await getMilestones();
+  const milestone = milestones.find((m) => m.title === milestoneTitle);
+  if (!milestone) {
+    logger.warn(
+      `Milestone '${milestoneTitle}' did not exists. Adding of milestone skipped.`,
+    );
+    return;
+  }
+
+  // then, add the milestone id to the PR
+  const repository = config.parentRepo ?? config.repository;
+  await githubApi.patchJson(`repos/${repository}/issues/${issueNo}`, {
+    body: {
+      milestone: milestone.number,
+    },
+  });
+}
+
+async function getMilestones(state = 'open'): Promise<Milestone[]> {
```

Comment:
This list needs to be cached and persisted. Is it likely that there's just one image of milestones? Ie <=100